src/algebra/big_operators/basic.lean

```diff
@@ -1718,6 +1718,48 @@ begin
   { rw [finset.sum_insert hat, finset.prod_insert hat, multiset.prod_add, ih] }
 end
 
+lemma disjoint_of_disjoint_sum {α β : Type*} {i : finset β} {f : β → multiset α} {z : β}
+  (h : ∀ x ∈ i, multiset.disjoint (f x) (f z)) :
+  multiset.disjoint (f z) (∑ x in i, f x) :=
+begin
+  classical,
+  apply finset.sum_induction f (λ (t : multiset α), disjoint (f z) t),
+  { exact λ _ _ ha hb, disjoint_add_right.mpr ⟨ha, hb⟩ },
+  { exact (zero_disjoint (f z)).symm},
+  { exact λ x hx, (h x hx).symm },
+end
+
+lemma le_of_disjoint_sum_le {α β : Type*} {T : multiset α} {i : finset β} {f : β → multiset α}
+  (h1 : ∀ x ∈ i, f x ≤ T) (h2 : ∀ x y ∈ i, x ≠ y → multiset.disjoint (f x) (f y)) :
+  ∑ x in i, f x ≤ T :=
```

Comment:
Note that this naming is backwards: `foo_of_bar` means `(h : bar) : foo`; so this should be called `sum_le_of_le_of_disjoint`

Reply:
Well, the thing is the result is not true as stated for `list` and `multiset` since they could contain the same `multiset` several times so I am not sure what the correct statement would be in those cases.

Reply:
I added a new function to help with the proof: 
```lean
lemma add_le_of_le_of_disjoint [decidable_eq α] {s t T : multiset α}
  (hs : s ≤ T) (ht : t ≤ T) (h : multiset.disjoint s t):
  s + t ≤ T 
```